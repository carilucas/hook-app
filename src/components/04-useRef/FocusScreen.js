import React, {useRef} from 'react';
import '../01-useState/counter.css';

export const FocusScreen = () => {

   const inputRef = useRef()

   const handleClick = ()=>{
      inputRef.current.select();
   }
   return (
      <div>
         <h1>Focus Screen</h1>
         <hr/>
         <input 
            ref={inputRef}
            type="text" 
            className="form-control" 
            placeholder="Nombre"
         />
         <button 
            className="btn btn-outline-primary mt-2"
            onClick={handleClick}
         >
            Focus
         </button>
      </div>
   )
}
