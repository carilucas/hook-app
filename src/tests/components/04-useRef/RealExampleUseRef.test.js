import { shallow } from "enzyme";
import { RealExampleUseRef } from "../../../components/04-useRef/RealExampleUseRef";
import '@testing-library/jest-dom';


describe('Pruebas de RealExampleUseRef', () => {

   const wrapper = shallow(<RealExampleUseRef/>);
   
   test('Debe mostrarse el componente RealExampleUseRef snapshot', () => {

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
      console.log(wrapper.html());
   });

   test('Debe mostrar el componente <MultipleCustomHooks/>', () => {
      
      wrapper.find('button').simulate('click');
      expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
      console.log(wrapper.html());



   })
   
   

})
