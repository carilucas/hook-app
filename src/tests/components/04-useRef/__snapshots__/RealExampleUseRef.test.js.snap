// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de RealExampleUseRef Debe mostrarse el componente RealExampleUseRef snapshot 1`] = `
<div>
  <h1>
    RealExampleUseRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
